services/issue/status.go

```diff
@@ -6,21 +6,14 @@ package issue
 import (
 	"context"
 
-	"code.gitea.io/gitea/models/db"
 	issues_model "code.gitea.io/gitea/models/issues"
 	user_model "code.gitea.io/gitea/models/user"
 	"code.gitea.io/gitea/modules/log"
 	"code.gitea.io/gitea/modules/notification"
 )
 
 // ChangeStatus changes issue status to open or closed.
-func ChangeStatus(issue *issues_model.Issue, doer *user_model.User, commitID string, closed bool) error {
-	return changeStatusCtx(db.DefaultContext, issue, doer, commitID, closed)
-}
-
-// changeStatusCtx changes issue status to open or closed.
-// TODO: if context is not db.DefaultContext we get a deadlock!!!
```

Comment:
What happened here?
Why can we do that now without getting a deadlock?

Reply:
I think the comment should be wrong. `db.DefaultContext` is never be a method to avoid deadlock.